backend/src/pong/pong.service.ts

```diff
@@ -95,24 +121,22 @@ export class PongService {
 		//  check collision for paddle one
 		if (this.doesHitPaddle(gameState.ball, paddleP1, gameState.canvas)) {
 			console.log('Collision Paddle one');
-			const paddleCenter = paddleP1.position.y + paddleP1.height / 2;
-			const d = paddleCenter - ballPos.y;
-			//  speed up the ball
-			this.dx = -(this.dx - 0.02);
-			// change bounce angle
-			this.dy -= d * -0.01;
-			// check collision for paddle two
-			// DRY I know but
+			this.changeBallDirection(paddleP1, gameState.ball.position.y, -1);
 		} else if (this.doesHitPaddle(gameState.ball, paddleP2, gameState.canvas)) {
 			console.log('Collision Paddle two');
-			const paddleCenter = paddleP2.position.y + paddleP2.height / 2;
-			const d = paddleCenter - ballPos.y;
-			// speed up ball
-			this.dx = -(this.dx + 0.02);
-			//  change bounce angle
-			this.dy += d * -0.01;
+			this.changeBallDirection(paddleP2, gameState.ball.position.y, 1);
 		} else if (this.doesHitWall(radius, ballPos.x, this.dx)) {
-			console.log('Point is over');
+			//  check which side the ball hit to decide who the winner is
+			ballPos.x > 80 ? ++gameState.scorePlayerOne : ++gameState.scorePlayerTwo;
```

Comment:
To check which which side of the canvas the impact happened 😆  So 80% of the canvas width (right side) Not the cleanest solution but yea...

Reply:
hehe yes okay, so that I got correct. Why 80% though? Why not 50%?
anyway, technically it's a "magic number", so I would be against it. But I also don't like that javascript doesn't actually offer a thing to do "real" defines like in C (`#define RIGHT_SIDE 80`) only const variables.. 
soooo blaaaa

i get it now
can resolve

Reply:
I agree, made `const midWayPoint = 50`
I used 80, because, because, reasons 🗡️